protocol/daemons/liquidation/client/client_test.go

```diff
@@ -241,245 +245,94 @@ func TestRunLiquidationDaemonTaskLoop(t *testing.T) {
-						Limit: df.Liquidation.SubaccountPageLimit,
-					},
-				}
-				response := &satypes.QuerySubaccountAllResponse{
-					Subaccount: []satypes.Subaccount{
-						constants.Carl_Num0_599USD,
-						constants.Dave_Num0_599USD,
-					},
-				}
-				mck.On("SubaccountAll", ctx, req).Return(response, nil)
-			},
-			expectedSubaccounts: []satypes.Subaccount{
-				constants.Carl_Num0_599USD,
-				constants.Dave_Num0_599USD,
-			},
-		},
-		"Success Paginated": {
-			setupMocks: func(ctx context.Context, mck *mocks.QueryClient) {
-				req := &satypes.QueryAllSubaccountRequest{
-					Pagination: &query.PageRequest{
-						Limit: df.Liquidation.SubaccountPageLimit,
-					},
-				}
-				nextKey := []byte("next key")
-				response := &satypes.QuerySubaccountAllResponse{
-					Subaccount: []satypes.Subaccount{
-						constants.Carl_Num0_599USD,
-					},
-					Pagination: &query.PageResponse{
-						NextKey: nextKey,
-					},
-				}
-				mck.On("SubaccountAll", ctx, req).Return(response, nil)
-				req2 := &satypes.QueryAllSubaccountRequest{
-					Pagination: &query.PageRequest{
-						Key:   nextKey,
-						Limit: df.Liquidation.SubaccountPageLimit,
-					},
-				}
-				response2 := &satypes.QuerySubaccountAllResponse{
-					Subaccount: []satypes.Subaccount{
-						constants.Dave_Num0_599USD,
-					},
-				}
-				mck.On("SubaccountAll", ctx, req2).Return(response2, nil)
-			},
-			expectedSubaccounts: []satypes.Subaccount{
-				constants.Carl_Num0_599USD,
-				constants.Dave_Num0_599USD,
-			},
-		},
-		"Errors are propagated": {
-			setupMocks: func(ctx context.Context, mck *mocks.QueryClient) {
-				req := &satypes.QueryAllSubaccountRequest{
-					Pagination: &query.PageRequest{
-						Limit: df.Liquidation.SubaccountPageLimit,
-					},
-				}
-				mck.On("SubaccountAll", ctx, req).Return(nil, errors.New("test error"))
-			},
-			expectedError: errors.New("test error"),
-		},
+func NewFakeSubTaskRunnerWithError(err error) *FakeSubTaskRunner {
+	return &FakeSubTaskRunner{
+		err: err,
 	}
+}
 
-	for name, tc := range tests {
-		t.Run(name, func(t *testing.T) {
-			queryClientMock := &mocks.QueryClient{}
-			tc.setupMocks(grpc.Ctx, queryClientMock)
-
-			actual, err := client.GetAllSubaccounts(grpc.Ctx, queryClientMock, df.Liquidation.SubaccountPageLimit)
-			if err != nil {
-				require.EqualError(t, err, tc.expectedError.Error())
-			} else {
-				require.Equal(t, tc.expectedSubaccounts, actual)
-			}
-		})
-	}
+// RunLiquidationDaemonTaskLoop is a mock implementation of the SubTaskRunner interface. It records the
+// call as a sanity check, and returns the error set by NewFakeSubTaskRunnerWithError.
+func (f *FakeSubTaskRunner) RunLiquidationDaemonTaskLoop(
+	_ context.Context,
+	_ flags.LiquidationFlags,
+	_ satypes.QueryClient,
+	_ clobtypes.QueryClient,
+	_ api.LiquidationServiceClient,
+) error {
+	f.called = true
+	return f.err
 }
 
-func TestCheckCollateralizationForSubaccounts(t *testing.T) {
+func TestHealthCheck_Mixed(t *testing.T) {
```

Comment:
This diff is a bit painful! The test methods on the left were migrated out of this file. TestHealthCheck is entirely new.